pinot-query-runtime/src/test/java/org/apache/pinot/query/runtime/QueryRunnerTest.java

```diff
@@ -200,6 +200,13 @@ private Object[][] provideTestSql() {
         new Object[]{"SELECT a.col1, b.col2 FROM a JOIN b ON a.col1 = b.col1 "
             + " WHERE a.col1 IN ('foo') AND b.col2 NOT IN ('')"},
 
+        // Range conditions with continuous and non-continuous range.
+        new Object[]{"SELECT a.col1, b.col2 FROM a JOIN b ON a.col1 = b.col1 "
+            + " WHERE a.col3 IN (1, 2, 3) OR (a.col3 > 10 AND a.col3 < 50)"},
```

Comment:
Can you also check the case where someone tries an impossible range condition or conditions with overlaps? I was running into an issue with those edge-cases in my PR. Examples:

```
(a.col3 > 10 AND a.col3 < 20) AND (a.col3 > 30 AND a.col3 < 40) # Impossible range
(a.col3 > 10 AND a.col3 < 20) AND (a.col3 > 15 AND a.col3 < 25) # Overlapping range
```

Reply:
yeah I was planning to split that into a separate PR but yeah sure let me create this one.

Reply:
is IN (<single value>) getting rewritten to = ?

FWIW, in the current engine we do this rewrite but not really worth it

Reply:
single IN yes will be rewrite